packages/ai/src/ai/common/chat.py

```
@@ -451,6 +478,86 @@ def chat(self, question: Question) -> Answer:
+        max_delay = CONST_CHAT_MAX_DELAY
+
+        for attempt in range(max_network_retries):
+            emitted = {'value': False}
+
+            def gated(text: str) -> None:
+                emitted['value'] = True
+                on_chunk(text)
+
+            try:
+                return self._chat_stream(prompt, gated)
+            except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)